src/docmanager/default.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IKernelSpecId
+} from 'jupyter-js-services';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  loadModeByFileName
+} from '../codemirror';
+
+import {
+  CodeMirrorWidget
+} from '../codemirror/widget';
+
+import {
+  IDocumentModel, IWidgetFactory, IModelFactory, IDocumentContext
+} from './index';
+
+
+/**
+ * The default implementation a document model.
+ */
+export
+class DocumentModel implements IDocumentModel {
+  /**
+   * Construct a new document model.
+   */
+  constructor(path: string, kernelSpecs: IKernelSpecId[]) {
+    // Use the path and the kernel specs to set the default
+    // kernel and language.
+  }
```

Comment:
In what cases do you see us using the kernel specs to set the default kernel? I was imagining for a new file, it would just do a translation from extension -> language for things like *.py files. For *.ipynb files, it wouldn't provide a default until the content was loaded and we could pull it from the .ipynb file.